p2p/bad_peer_test.go

```diff
@@ -0,0 +1,154 @@
+package p2p_test
+
+import (
+	"context"
+	"encoding/binary"
+	"path/filepath"
+	"testing"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	pb "github.com/libp2p/go-libp2p-pubsub/pb"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	varint "github.com/multiformats/go-varint"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/cmd/node"
```

Comment:
i don't think that we should import node here. is it possible to write a test without a node, or otherwise add this test to a node module

Reply:
We can add it to the node module, but what is the problem you see with importing node here?

Reply:
you will see a cycle if you want to import this module in the tests for node

and even conceptually, there should be a hierarchy of imports in the project. higher level things (as node) should not be imported by lower level things (such as encoding, p2p, database)

Reply:
With regards to the hierarchy totally agree with respect to production code. But for tests I think this can be a good approach.

Consider a situation where you want to test an API and you wan't to test it with the full stack active, I.E not mocking out any stuff so having a full node running. You could put these tests in node, but does it really make sense to put the tests in node when they are testing an API that is defined in a different package? If you do that for all the APIs the node package will end up with a lot of tests that don't really have anything to do with node. Structurally there is no problem with importing node in test packages because you can't depend on a test package.

> i think it will also lead to failures in p2p module, because of some changes in node. in my opinion thats a mess

I don't see this as a problem, if node is not supporting correct functioning of the API then that's a real problem. For example if the API is not registered in node then it's not going to work, that is something we want to know about.

Reply:
I've moved the test to the node package.